justifi-tech/web-component-library
# feat: add font customization

Developer considerations
--------------

  - If this is a MAJOR version, what are the breaking API changes? Was anything added to the API?
  - If this is a MINOR version, what feature was added? Why would consumers want this new version?
  - If this is a PATCH version, what was fixed?


Steps for Testing/QA
--------------------

This doesn't require integration with any service, but to test it, the branch `FE-196-find-a-way-to-add-matching-font-to-components` needs to be used in both `web-component-library` repo and `payments-js` repo.

The following should be expected:
- Components that are styled through CSS variables use the `--jfi-load-google-font` to load the appropriate fonts in the appropriate contexts (Within the iFrame and outside of it for mixed components like `PaymentForm`).
- Loaded fonts are appropriately applied with the `--jfi-layout-font-family` and `--jfi-form-label-font-family` variables.

Screenshots:
<img width="1026" alt="image" src="https://github.com/justifi-tech/web-component-library/assets/10523683/5f3fb190-5a6a-4669-b0b1-b41f7aeeadf2">
<img width="1028" alt="image" src="https://github.com/justifi-tech/web-component-library/assets/10523683/fcd23e1a-5057-4b7e-894b-8a7e6490e097">
<img width="1029" alt="image" src="https://github.com/justifi-tech/web-component-library/assets/10523683/095d28ae-f15b-4ceb-86a7-ac2cccb7caf8">
